prow/pjutil/pjutil_test.go

```diff
@@ -114,71 +115,8 @@ func TestProwJobToPod(t *testing.T) {
 		if err != nil {
 			t.Errorf("unexpected error: %v", err)
 		}
-		// TODO: For now I am just comparing fields manually, eventually we
-		// should port the semantic.DeepEqual helper from the api-machinery
-		// repo, which is basically a fork of the reflect package.
-		// if !semantic.DeepEqual(got, test.expected) {
-		//	 t.Errorf("got pod:\n%#v\n\nexpected pod:\n%#v\n", got, test.expected)
-		// }
-		var foundCreatedByLabel, foundTypeLabel, foundJobAnnotation bool
-		for key, value := range got.ObjectMeta.Labels {
-			if key == kube.CreatedByProw && value == "true" {
-				foundCreatedByLabel = true
-			}
-			if key == kube.ProwJobTypeLabel && value == string(pj.Spec.Type) {
-				foundTypeLabel = true
-			}
-			var match bool
-			for k, v := range test.expected.ObjectMeta.Labels {
-				if k == key && v == value {
-					match = true
-					break
-				}
-			}
-			if !match {
-				t.Errorf("expected labels: %v, got: %v", test.expected.ObjectMeta.Labels, got.ObjectMeta.Labels)
-			}
-		}
-		for key, value := range got.ObjectMeta.Annotations {
-			if key == kube.ProwJobAnnotation && value == pj.Spec.Job {
-				foundJobAnnotation = true
-			}
-		}
-		if !foundCreatedByLabel {
-			t.Errorf("expected a created-by-prow=true label in %v", got.ObjectMeta.Labels)
-		}
-		if !foundTypeLabel {
-			t.Errorf("expected a %s=%s label in %v", kube.ProwJobTypeLabel, pj.Spec.Type, got.ObjectMeta.Labels)
-		}
-		if !foundJobAnnotation {
-			t.Errorf("expected a %s=%s annotation in %v", kube.ProwJobAnnotation, pj.Spec.Job, got.ObjectMeta.Annotations)
-		}
-
-		expectedContainer := test.expected.Spec.Containers[i]
-		gotContainer := got.Spec.Containers[i]
-
-		dumpGotEnv := false
-		for _, expectedEnv := range expectedContainer.Env {
-			found := false
-			for _, gotEnv := range gotContainer.Env {
-				if expectedEnv.Name == gotEnv.Name && expectedEnv.Value == gotEnv.Value {
-					found = true
-					break
-				}
-			}
-			if !found {
-				dumpGotEnv = true
-				t.Errorf("could not find expected env %s=%s", expectedEnv.Name, expectedEnv.Value)
-			}
-		}
-		if dumpGotEnv {
-			t.Errorf("expected env:\n%#v\ngot:\n%#v\n", expectedContainer.Env, gotContainer.Env)
-		}
-		if expectedContainer.Image != gotContainer.Image {
-			t.Errorf("expected image: %s, got: %s", expectedContainer.Image, gotContainer.Image)
-		}
-		if test.expected.Spec.RestartPolicy != got.Spec.RestartPolicy {
-			t.Errorf("expected restart policy: %s, got: %s", test.expected.Spec.RestartPolicy, got.Spec.RestartPolicy)
+		if !equality.DeepEqual(got, test.expected) {
```

Comment:
this doesn't match what we have vendored, not sure about upstream

Reply:
fixed

The test also needs fixing

/hold